libcontainer/nsenter/nsexec.c

```diff
@@ -681,6 +715,11 @@ void nsexec(void)
 				 * clone_parent rant). So signal our parent to hook us up.
 				 */
 
+				/* Switching is only necessary if we joined namespaces. */
+				if (config.namespaces) {
+					if (prctl(PR_SET_DUMPABLE, 1, 0, 0, 0) < 0)
+						bail("failed to set process as dumpable");
+				}
```

Comment:
`/proc/self/uid_map` and `/proc/self/gid_map` become root-owned if you're not dumpable. So the process doing mapping doesn't have privileges to do the mapping.